js/src/widgets.js

```diff
@@ -73,19 +76,35 @@ jQuery(
 			wp.customize.control.each( customize_add_flag );
 			wp.customize.control.bind( 'add', customize_add_flag );
 
+		} else if ( isBlockEditor ) {
+
+			widgets_container = $( '.edit-widgets-main-block-list' );
+			widgets_selector  = '.widget';
+
 		} else {
 
 			widgets_container = $( '#widgets-right' );
 			widgets_selector  = '.widget';
 
 		}
 
-		// Add flags on load.
-		$( widgets_selector, widgets_container ).each(
-			function() {
-				add_flag( this );
-			}
-		);
+		if ( ! isBlockEditor ) {
+			// Add flags on load.
+			$( widgets_selector, widgets_container ).each(
+				function() {
+					add_flag( this );
+				}
+			);
+		} else {
+			// Update flags when we click on the legacy widget to display its form.
+			widgets_container.on(
+				'click',
+				'.wp-block-legacy-widget',
+				function() {
+					add_flag( $( this ).find( '.widget' ) );
+				}
+			);
```

Comment:
Couldn't this be done here : https://github.com/polylang/polylang/blob/f5afeec0da1faf6b530f8dca50027a9099a74509/js/src/widgets.js#L79-L84